test/jdk/java/awt/FileDialog/DoubleActionESC.java

```diff
@@ -51,42 +53,40 @@ public class DoubleActionESC {
     private static volatile Dimension d;
     private static volatile CountDownLatch latch;
     private static final int REPEAT_COUNT = 2;
+    private static final long LATCH_TIMEOUT = 4;
 
     public static void main(String[] args) throws Exception {
-        latch = new CountDownLatch(1);
+        latch = new CountDownLatch(REPEAT_COUNT);
 
         robot = new Robot();
-        robot.setAutoDelay(100);
+        robot.setAutoDelay(50);
         try {
             EventQueue.invokeAndWait(() -> {
                 createAndShowUI();
             });
 
+            robot.waitForIdle();
             robot.delay(1000);
+
             EventQueue.invokeAndWait(() -> {
                 p = showBtn.getLocationOnScreen();
                 d = showBtn.getSize();
             });
 
             for (int i = 0; i < REPEAT_COUNT; ++i) {
-                Thread thread = new Thread(() -> {
-                    robot.mouseMove(p.x + d.width / 2, p.y + d.height / 2);
-                    robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
-                    robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
-                });
-                thread.start();
-                robot.delay(3000);
+                robot.mouseMove(p.x + d.width / 2, p.y + d.height / 2);
+                robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+                robot.waitForIdle();
+                robot.delay(1000);
 
-                Thread thread1 = new Thread(() -> {
-                    robot.keyPress(KeyEvent.VK_ESCAPE);
-                    robot.keyRelease(KeyEvent.VK_ESCAPE);
-                    robot.waitForIdle();
-                });
-                thread1.start();
-                robot.delay(3000);
+                robot.keyPress(KeyEvent.VK_ESCAPE);
+                robot.keyRelease(KeyEvent.VK_ESCAPE);
+                robot.waitForIdle();
+                robot.delay(1000);
             }
 
-            latch.await();
+            latch.await(LATCH_TIMEOUT, SECONDS);
             if (fd.isVisible()) {
```

Comment:
I'm on the fence here… Calling `fd.isVisible()` on EDT is more reliable; but it's not strictly required for AWT components.